### PR TITLE
Fix #334, Add Functional Tests Step to CI

### LINKED ISSUE #334: Add functional tests step to continuous integration
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [x] I reviewed the [cFS README.md file](https://github.com/nasa/cFS/blob/main/README.md) to see if the feature is in the major future work.
* [x] I performed a cursory search to see if the feature request is relevant, not redundant, nor in conflict with other tickets.

**Is your feature request related to a problem? Please describe.**

The cFE repository has a suite of functional tests. These tests can sometimes fail when introducing new osal dependencies. These failures can be shown to be fixed at the bundle level when combining the right set of submodule references. 

**Describe the solution you'd like**
Port the functional test workflow as an extra job to the build test and run workflows. 

**Describe alternatives you've considered**
Make a separate workflow

**Additional context**
None

### PR DESCRIPTION
**Pull request by ArielSAdamsNASA**
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [ ] I signed and emailed the appropriate Contributor License agreement to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
Fixes #334 

**Testing performed**
Tested on fork repo. 
Deprecated Build, Test, and Run: https://github.com/ArielSAdamsNASA/cFS-JSF-Rules/actions/runs/1198344779
Build, Test, and Run [OMIT_DEPRECATED=true]: https://github.com/ArielSAdamsNASA/cFS-JSF-Rules/actions/runs/1198344780
Also seen in checks in this PR. 

**Expected behavior changes**
Build, test, and run workflows runs functional tests. Uses cache to reduce build time. 

**Contributor Info - All information REQUIRED for consideration of pull request**
Ariel Adams, ASRC Federal

### CODE REVIEW

**Comment by skliper** [comment]
Looks like 4 jobs that really could be consolidated?  Build, make test, then run with the functional test app.  Or if these really need to be separate could the build be archived so it doesn't actually need to be built 4 times?  Currently may be burning a lot of minutes w/ the duplicated work (especially since this is matrixed).

File: .github/workflows/build-cfs-deprecated.yml [hunk 1]
```diff
@@ -161,3 +162,74 @@ jobs:
                   exit -1
           fi
         working-directory: ./build/exe/cpu1/
+
+  Local-Test-Build:
```
Comment (astrogeco) [inline .github/workflows/build-cfs-deprecated.yml:166]: This should have a new name

File: .github/workflows/build-cfs.yml [hunk 2]
```diff
@@ -159,3 +160,74 @@ jobs:
                   exit -1
           fi
         working-directory: ./build/exe/cpu1/
+
+  Local-Test-Build:
```
Comment (astrogeco) [inline .github/workflows/build-cfs.yml:164]: Rename

**Review by astrogeco — APPROVED** [review]
I'm fine with the separate jobs. We can figure out how to reuse the build files using cache. The one thing i would add is a dependency between run and functional test so that we don't even try to run the functional tests if plain run ./core-cpu doesn't work

File: .github/workflows/build-cfs-deprecated.yml (hunk 1 above)
Comment (ArielSAdamsNASA) [reply .github/workflows/build-cfs-deprecated.yml]: Updated to Functional-Test-Build

File: .github/workflows/build-cfs.yml (hunk 2 above)
Comment (ArielSAdamsNASA) [reply .github/workflows/build-cfs.yml]: Updated to Functional-Test-Build

**Pull request merged**